plugins/kubernetes/app/decorators/project_decorator.rb

```diff
@@ -1,4 +1,64 @@
 Project.class_eval do
   has_many :kubernetes_releases, through: :builds, class_name: 'Kubernetes::Release'
   has_many :roles, class_name: 'Kubernetes::Role'
+
+  def file_from_repo(path, git_ref, ttl: 1.hour)
+    # Caching by Git reference
+    Rails.cache.fetch([git_ref, path], expire_in: ttl) do
+      data = GITHUB.contents(github_repo, path: path, ref: git_ref)
+      Base64.decode64(data[:content])
+    end
+  rescue Octokit::NotFound
+    nil
+  end
+
+  def directory_contents_from_repo(path, git_ref, ttl: 1.hour)
+    # Caching by Git reference
+    Rails.cache.fetch([git_ref, path], expire_in: ttl) do
+      GITHUB.contents(github_repo, path: path, ref: git_ref).map(&:path).select { |file_name|
+        file_name.ends_with?('.yml', '.yaml', '.json')
+      }
+    end
+  rescue Octokit::NotFound
+    nil
+  end
+
+  # Imports the new kubernetes roles. This operation is atomic: if one role fails to be imported, none
+  # of them will be persisted and the soft deletion will be rollbacked.
+  def refresh_kubernetes_roles!(git_ref)
+    config_files = directory_contents_from_repo('kubernetes', git_ref)
+
+    if config_files.try(:any?)
```

Comment:
Again, just use ```empty?```, quicker as it's just a size check, any? will allow nils to exist possibly causing errors further down.